scripts/openyurt-deployer/main.go

```diff
@@ -0,0 +1,260 @@
+// MIT License
+//
+// Copyright (c) 2023 Jason Chua, Ruiqi Lai and vHive team
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"os"
+	"strings"
+
+	"github.com/vhive-serverless/vHive/scripts/utils"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type NodesInfo struct {
+	Master  string  `json:"master"`
+	Workers Workers `json:"workers"`
+}
+
+type Workers struct {
+	Cloud []string `json:"cloud"`
+	Edge  []string `json:"edge"`
+}
+
+func main() {
+	if len(os.Args) != 2 {
+		utils.InfoPrintf("Usage: %s  <operation: deploy | clean | demo-c | demo-e | demo-clear | demo-print> [Parameters...]\n", os.Args[0])
+		os.Exit(-1)
+	}
+
+	var (
+		deployerConf = flag.String("conf", "conf.json",
+			`Configuration file with the following structure:
+			{
+				"master": "user@master",
+				"workers": {
+					"cloud": [
+						"user@cloud-0"
+					],
+					"edge": [
+						"user@edge-0"
+					]
+				}
+			}
+			`)
+		logLvl = flag.String("loglvl", "debug", "Debug level: 'info' or 'debug'")
+	)
+	flag.Parse()
+	log.SetOutput(os.Stdout)
+	switch *logLvl {
+	case "info":
+		log.SetLevel(log.InfoLevel)
+	case "debug":
+		log.SetLevel(log.DebugLevel)
+		log.Debug("Debug mode is enabled")
+	default:
+		log.Fatalf("Invalid log level: '%s', expected 'info' or 'debug'", *logLvl)
+	}
+	operation := os.Args[1]
+	switch operation {
+	case "k8s":
+		deployNodes(*deployerConf)
+	case "knative":
+		deployKnative(*deployerConf)
+	// case "clean":
+	// 	cleanNodes(*deployerConf)
+	// case "demo-e":
+	// 	demo(*deployerConf, false)
+	// case "demo-c":
+	// 	demo(*deployerConf, true)
+	// case "demo-clear":
+	// 	delDemo(*deployerConf)
+	// case "demo-print":
+	// 	printDemo(*deployerConf)
```

Comment:
are these modes supported? the readme mentions them

Reply:
yes, edited again in readme